SCons/Node/Python.py

```diff
@@ -138,24 +138,20 @@ def read(self):
     def get_text_contents(self):
         """By the assumption that the node.built_value is a
         deterministic product of the sources, the contents of a Value
-        are the concatenation of all the contents of its sources.  As
-        the value need not be built when get_contents() is called, we
+        are the concatenation of all the content signatures of its sources.
+        As the value need not be built when get_contents() is called, we
         cannot use the actual node.built_value."""
-        ###TODO: something reasonable about universal newlines
         contents = str(self.value)
-        for kid in self.children(None):
-            contents = contents + kid.get_contents().decode()
+        contents += ''.join(n.get_csig() for n in self.children())
```

Comment:
As I've said before I'm not sure concatenating the contents of the child was correct, and this seems even less correct.